src/object_gui.cpp

```diff
@@ -530,5 +661,5 @@ void ShowBuildObjectPicker()
 /** Reset all data of the object GUI. */
 void InitializeObjectGui()
 {
-	_selected_object_class = (ObjectClassID)0;
+	// No-op.
```

Comment:
Is it possible to remove the function InitializeObjectGui?
Anyway, it is strange you can remove it. I guess when you are playing a game and then load savegame, you should reset the object GUI.

Reply:
With this change I am not resetting the selected object since I want the user to be able to open the objects GUI somewhere and, if and only if that object is still loaded (the corresponding NewGRF) then the user will get that selected first.
I you strongly believe it should get reset to nullptr I am happy to update it.

Reply:
Brough it back as setting the _selected_object_class to nullptr.